compat/db.js

```diff
@@ -13,21 +14,21 @@ exports.manifest = {
 }
 
 exports.init = function (sbot, config) {
-  sbot.add = sbot.db.add
-  sbot.get = function get(idOrObject, cb) {
+  sbot.add = Hookable(sbot.db.add)
+  sbot.get = Hookable(function get(idOrObject, cb) {
     if (typeof idOrObject === 'object' && idOrObject.meta) {
       sbot.db.getMsg(idOrObject.id, cb)
     } else if (typeof idOrObject === 'object') {
       sbot.db.get(idOrObject.id, cb)
     } else {
       sbot.db.get(idOrObject, cb)
     }
-  }
-  sbot.publish = sbot.db.publish
-  sbot.whoami = () => ({ id: sbot.id })
-  sbot.ready = () => true
+  })
+  sbot.publish = Hookable(sbot.db.publish)
```

Comment:
sbot.db.publish is created by the publish compat plugin. i wonder if this throws if we don't load that first?

Reply:
made a new PR with some changes, didn't want to juggle the branches around https://github.com/ssbc/ssb-db2/pull/427